scripts/gitlab/lib.sh

```diff
@@ -96,3 +96,79 @@ curl -XPOST -d "$1" "https://matrix.parity.io/_matrix/client/r0/rooms/$2/send/m.
 # Pretty-printing functions
 boldprint () { printf "|\n| \033[1m%s\033[0m\n|\n" "${@}"; }
 boldcat () { printf "|\n"; while read -r l; do printf "| \033[1m%s\033[0m\n" "${l}"; done; printf "|\n" ; }
+
+prepare_git() {
+  # Set the user name and email to make merging work
+  git config --global user.name 'CI system'
+  git config --global user.email '<>'
+}
+
+prepare_substrate() {
+  pr_companion=$1
+  boldprint "companion pr specified/detected: #${pr_companion}"
+
+  # Clone the current Substrate master branch into ./substrate.
+  # NOTE: we need to pull enough commits to be able to find a common
+  # ancestor for successfully performing merges below.
+  git clone --depth 20 https://github.com/paritytech/substrate.git
+  previous_path=$(pwd)
+  cd substrate
+  SUBSTRATE_PATH=$(pwd)
+
+  git fetch origin refs/pull/${pr_companion}/head:pr/${pr_companion}
+  git checkout pr/${pr_companion}
+  git merge origin/master
+
+  cd "$previous_path"
+
+  # Merge master into our branch before building Polkadot to make sure we don't miss
+  # any commits that are required by Polkadot.
+  git merge origin/master
+
+  # Make sure we override the crates in native and wasm build
+  # patching the git path as described in the link below did not test correctly
+  # https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html
+  mkdir .cargo
+  echo "paths = [ \"$SUBSTRATE_PATH\" ]" > .cargo/config
+
+  mkdir -p target/debug/wbuild/.cargo
+  cp .cargo/config target/debug/wbuild/.cargo/config
+}
+
+pull_companion_substrate() {
+  github_api_polkadot_pull_url="https://api.github.com/repos/paritytech/polkadot/pulls"
+  # use github api v3 in order to access the data without authentication
+  github_header="Authorization: token ${GITHUB_PR_TOKEN}"
+
+
+  # either it's a pull request then check for a companion otherwise use
+  # substrate:master
+  if expr match "${CI_COMMIT_REF_NAME}" '^[0-9]\+$' >/dev/null
+  then
+    boldprint "this is pull request no ${CI_COMMIT_REF_NAME}"
+
+    pr_data_file="$(mktemp)"
+    # get the last reference to a pr in substrate
+    curl -sSL -H "${github_header}" -o "${pr_data_file}" \
+      "${github_api_polkadot_pull_url}/${CI_COMMIT_REF_NAME}"
+
+    pr_body="$(sed -n -r 's/^[[:space:]]+"body": (".*")[^"]+$/\1/p' "${pr_data_file}")"
```

Comment:
```suggestion
    pr_body="$(jq '.body' < ${pr_data_file})"
```
I'm fairly sure `jq`'s in basically all of our CI images, and is much more reliable for parsing JSON than `sed`.